File: Piko/Sources/SFStaticNode.m
//
//  SFStaticNode.m
//
//  Copyright (c) 2014 Satoshi Muraki. All rights reserved.
//

#import "SFStaticNode.h"

@implementation SFStaticNode

- (instancetype)initWithValue:(double)value
{
    SFOutput *output = [[SFOutput alloc] initWithName:@"output"];
    self = [super initWithInputs:@[] outputs:@[output]];
    if (self != nil) {
        _output = output;
        _value = value;
    }
    return self;
}

- (BOOL)rendersConstantFrame
{
    return YES;
}

- (BOOL)renderForPlayer:(SFPlaybackManager *)player
                  index:(UInt64)index
                  count:(NSInteger)count
                     to:(double *)buffer
{
    for (NSInteger i = 0; i < count; i++) {
        buffer[i] = _value;
    }
    return YES;
}

@end

File: Piko/Sources/SFStaticNode.swift
//
//  SFStaticNode.swift
//
//  Copyright (c) 2014 Satoshi Muraki. All rights reserved.
//

import Foundation

@objc class SFStaticNode: SFNode {

    let output: SFOutput

    var value: Double

    init(value: Double) {
        self.value = value
        output = SFOutput(name: "output")
        super.init(inputs: [], outputs: [output])
    }

    override func rendersConstantFrame() -> Bool {
        return true
    }

    override func renderForPlayer(player: SFPlaybackManager,
                                   index: UInt64,
                                   count: Int,
                               to buffer: UnsafeMutablePointer<Double>) -> Bool {
        for i in 0..<count {
            buffer[i] = value
        }
        return true
    }
}
